src/lib/plugins/login.js

```diff
@@ -1,28 +1,26 @@
 /* global BigInt */
 const Vec3 = require('vec3').Vec3
 
-const path = require('path')
 const crypto = require('crypto')
-const requireIndex = require('../requireindex')
-const plugins = requireIndex(path.join(__dirname, '..', 'plugins'))
 const playerDat = require('../playerDat')
 const convertInventorySlotId = require('../convertInventorySlotId')
+const plugins = require('./index')
 
 module.exports.server = function (serv, options) {
   serv._server.on('connection', client =>
     client.on('error', error => serv.emit('clientError', client, error)))
 
   serv._server.on('login', async (client) => {
-    if (client.socket.listeners('end').length === 0) return // TODO: should be fixed properly in nmp instead
+    if (client.socket?.listeners('end').length === 0) return // TODO: should be fixed properly in nmp instead
```

Comment:
socket is not defined when using a custom server: like in https://github.com/zardoy/prismarine-web-client/blob/318847186b76882989abc996527804d15785635d/src/customServer.js (don't see a reason to mock it instead)